docs/proposals/envvar-configmap.md

```diff
@@ -0,0 +1,162 @@
+# ConfigMaps as environment variables
+
+## Abstract
+
+A proposal for populating environment variables of a container from an entire ConfigMap.
+
+## Proposed Design
+
+Containers can specify the ConfigMaps that are consumed as environment variables.
+Each key defined in the ConfigMap's `Data` object must be a "C" identifier.
+
+Environment variables are currently defined by servies or the `Env` object in a container, where the `Env` takes precedence.
+The introduction of ConfigMaps adds a third possibility. To prevent any change in behavior, the `Env` object will still override any environment variable introduced by a ConfigMap. A ConfigMap is allowed to override variables defined by services.
+Variable references defined by an `EnvVar` struct can be resolved by values defined in other `EnvVar`s or those introduced by ConfigMaps and services.
+
+To prevent collisions amongst multiple ConfigMaps, each defined ConfigMap can have an associated prefix that is appended to each key in that ConfigMap. This also makes it obvious to the user where a given environment variable definition came from.
+
+### Kubectl updates
+
+The describe command will display the configmap name that have been defined as part of the environment variable section.
+
+### API Resource
+
+A new `EnvFromSource` type containing a `ConfigMapRef` will be added to the `Container` struct.
+
+```go
+// EnvFromSource represents the source of a set of ConfigMaps
+type EnvFromSource struct {
+  // A string to place in front of every key
+  Prefix string
```

Comment:
See the last section of the proposed design. The prefix is to prevent collision of keys amongst multiple configmap and to make it obvious to the user where a given environment variable was derived from.